pkg/pillar/cmd/zedmanager/handleresolve.go

```diff
@@ -0,0 +1,162 @@
+	c, _ := pub.Get(key)
+	if c == nil {
+		log.Infof("lookupResolveConfig(%s) not found\n", key)
+		return nil
+	}
+	config := c.(types.ResolveConfig)
+	log.Infof("lookupResolveConfig(%s) Done\n", key)
+	return &config
+}
+
+func lookupResolveStatus(ctx *zedmanagerContext,
+	key string) *types.ResolveStatus {
+
+	log.Infof("lookupResolveStatus(%s)\n", key)
+	sub := ctx.subAppImgResolveStatus
+	c, _ := sub.Get(key)
+	if c == nil {
+		log.Infof("lookupResolveStatus(%s) not found\n", key)
+		return nil
+	}
+	status := c.(types.ResolveStatus)
+	log.Infof("lookupResolveStatus(%s) Done\n", key)
+	return &status
+}
+
+func handleResolveStatusDelete(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	log.Infof("handleResolveStatusDelete for %s\n", key)
+	ctx := ctxArg.(*zedmanagerContext)
+	status := statusArg.(types.ResolveStatus)
+	config := lookupResolveConfig(ctx, status.Key())
+	if config != nil {
+		log.Infof("handleResolveStatusDelete delete config for %s\n",
+			key)
+		unpublishResolveConfig(ctx, config)
+	}
+	log.Infof("handleResolveStatusDelete done for %s\n", key)
+}
+
+func handleResolveStatusModify(ctxArg interface{}, key string,
+	statusArg interface{}) {
+
+	log.Infof("handleResolveStatusModify for %s\n", key)
+	ctx := ctxArg.(*zedmanagerContext)
+	rs := statusArg.(types.ResolveStatus)
+	pub := ctx.pubAppInstanceStatus
+	items := pub.GetAll()
+	for _, st := range items {
+		status := st.(types.AppInstanceStatus)
+		config := lookupAppInstanceConfig(ctx, status.Key())
+		if config == nil {
+			errStr := fmt.Sprintf("App Instance config not found for %s while resolving tags\n",
+				status.Key())
+			log.Errorf(errStr)
+			status.SetError(errStr, agentName, time.Now())
+			publishAppInstanceStatus(ctx, &status)
+			continue
+		}
+		for i := range status.StorageStatusList {
+			ss := &status.StorageStatusList[i]
+			if !ss.HasResolverRef || ss.Name != rs.Name || ss.DatastoreID != rs.DatastoreID {
+				continue
+			}
+			log.Infof("Updating SHA of storage status (%v) in app instance (%v)\n",
+				ss.ImageID, status.UUIDandVersion.UUID)
+			if len(rs.Error) != 0 {
+				errStr := fmt.Sprintf("Error occurred while resolving tags, updating app instance %s.\n",
+					status.Key())
+				log.Errorf(errStr)
+				status.SetError(rs.Error, rs.ErrorSource, rs.ErrorTime)
+				publishAppInstanceStatus(ctx, &status)
+				continue
+			}
+			if rs.ImageSha256 == "" {
+				errStr := fmt.Sprintf("Empty SHA in resolve status, updating app instance %s.\n",
+					status.Key())
+				log.Errorf(errStr)
+				status.SetError(errStr, agentName, time.Now())
+				publishAppInstanceStatus(ctx, &status)
+				continue
+			}
+			sc := lookupStorageConfig(config, *ss)
+			if sc == nil {
+				errStr := fmt.Sprintf("Storage config (%v) not found for app instance (%s) while resolving tags\n",
+					ss.ImageID, status.Key())
+				log.Errorf(errStr)
+				status.SetError(errStr, agentName, time.Now())
+				publishAppInstanceStatus(ctx, &status)
+				continue
```

Comment:
Why do you care about the StorageStatus? Sure, there should be one, but since you don't do anything with it, why even look?

Reply:
Sorry, I see the updateFromStorageConfig below. So its fine

Reply:
Okay, thanks.